### PR TITLE
Benchmark for ZE

### PR DESCRIPTION
**Pull request by zezevazquez**
#cloistered-gulls benchmark 15 exercises

### CODE REVIEW

File: src/guessTheNumber.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  } else if(Number.isInteger(input) && input > randomNum) {
+    return 'You too high!!!'
+  } else if(input === randomNum) {
+    return 'Yeee you win!!!'
+    process.exit(0)
+  } else if(input.toLowerCase() === 'exit'){
+    process.stdout.write('game over! way to quit')
+    process.exit(0)
+  } else if(!input || isNaN(input)) {
+    return 'Insert only integers RAWR!!!'
```
Comment (sdweber422) [inline src/guessTheNumber.js:14]: I think you could have wrapped the preceding if statements inside of this with an additional check for Number.isInteger(input). This would have avoided duplicate code and left the if-statements a little more readable.

File: src/guessTheNumber.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const randomNum = Math.floor(Math.random() * 100)
```
Comment (sdweber422) [inline src/guessTheNumber.js:1]: Could randomNum been part of the guessTheNumber function. It probably should if it is important to that function and would not affect implementation.

File: src/scramBag.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  'T':6,
+  'U':4,
+  'V':2,
+  'W':2,
+  'X':1,
+  'Y':2,
+  'Z':1,
+  '_':2
+  }
+  for(let i in input) {
```
Comment (sdweber422) [inline src/scramBag.js:31]: I would have converted input into uppercase, just in case the input was not entered that way. Also, would have checked for valid input.

File: src/scramBag.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      bagOfTiles[input[i]]--
+  }
+  return bagOfTiles
```
Comment (sdweber422) [inline src/scramBag.js:34]: There is not a function to return the values in the format specified in the README.

File: src/stringIndices.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export const stringIndices = (string, index) => {
+  const arrString = string.split(' ')
+  if (arrString.length < index || index < 0) {
+    return ''
```
Comment (sdweber422) [inline src/stringIndices.js:4]: Index in the specs would have started at 1. I do not see where that is accounted for in the code.

File: src/stringIndices.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export const stringIndices = (string, index) => {
+  const arrString = string.split(' ')
```
Comment (sdweber422) [inline src/stringIndices.js:2]: An error might be thrown for an empty string. Also, extra spaces would cause an error in obtaining the correct index. The following string 'Hello  there', has two spaces and the split method would cause an array to be formed in the manner['Hello', ' ', 'there'].

File: src/disemvowel.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export const disemvowel = (string) => {
+  const filteredString = string.toLowerCase().replace(/[aeiou]|\d|\W|\_/g, '')
```
Comment (sdweber422) [inline src/disemvowel.js:2]: This may cause an error if string is undefined or null.

File: tests/stringIndices_test.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  it('returns value of index', () => {
+    expect(stringIndices('Zeze loves dim sum', 3)).to.equal('sum')
+  })
+  it('returns empty string if calling on index that is greater than randomString.length', () => {
+    expect(stringIndices('Moar noodz por favor!', 7)).to.equal('')
+  })
+  it('returns empty string if calling on negative index', () => {
+    expect(stringIndices('Yaddadadadamean?', -2)).to.equal('')
+  })
+})
```
Comment (sdweber422) [inline tests/stringIndices_test.js:17]: In general, your tests should include more edge cases and be more robust. An interface of some type may have helped point this out.
